src/Workspaces/Core/Portable/Workspace/Solution/TextDocumentStates.cs

```diff
@@ -93,10 +93,8 @@ public TState GetRequiredState(DocumentId documentId)
         /// <returns></returns>
         public IEnumerable<TState> GetStatesInCompilationOrder()
         {
-            foreach (var id in Ids)
-            {
-                yield return _map[id];
-            }
+            var map = _map;
+            return Ids.Select(id => map[id]);
```

Comment:
intentional chance, this allows the runtime to produce an IEnumerable with a known length, which helps with any operations it does where it wants to allocate space.